arch/arm/include/syscall.h

```diff
@@ -136,6 +136,18 @@
 
 static inline uintptr_t sys_call0(unsigned int nbr)
 {
+#ifdef __ghs__
```

Comment:
have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first

Reply:
yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request

Reply:
> have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first

yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request

Reply:
> > have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first
> 
> yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request

which pull request?

Reply:
> > > have you enable "-gnu99" and "--gnu_asm" in command line ? I think you would better to upstream the GHS toolchain support first
> > 
> > 
> > yes, the "-gnu99" and "--gnu_asm" is enabled in another pull request
> 
> which pull request?

is in https://github.com/apache/nuttx/pull/12877 new pull request